metadata-ingestion/src/datahub/ingestion/source/hex/hex.py

```diff
@@ -93,9 +106,72 @@ class HexSourceConfig(
         default=True,
         description="Set ownership identity from owner/creator email",
     )
+    include_lineage: bool = Field(
+        default=True,
+        description='Include Hex lineage, being fetched from DataHub. See "Limitations" section in the docs for more details about the limitations of this feature.',
+    )
+    lineage_start_time: Optional[Union[str, datetime]] = Field(
+        default=None,
+        description="Earliest date of lineage to consider. Default: 1 day before lineage end time. You can specify absolute time like '2023-01-01' or relative time like '-7 days' or '-7d'.",
+    )
+    lineage_end_time: Optional[Union[str, datetime]] = Field(
+        default=None,
+        description="Latest date of lineage to consider. Default: Current time in UTC. You can specify absolute time like '2023-01-01' or relative time like '-1 day' or '-1d'.",
+    )
+    datahub_page_size: int = Field(
+        default=DATAHUB_API_PAGE_SIZE_DEFAULT,
+        description="Number of items to fetch per DataHub API call.",
+    )
 
+    def _set_lineage_times(cls, data: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
let's use a pydantic validator for this

Reply:
I tried at some moment and failed; `root_validator` is what I missed at that time

addressed in commit e797a71dbd9fe151019c321640efc7b070bbf58f